src/update/UpdateManager.cpp

```diff
@@ -140,6 +140,14 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
                         {
                             mAudacityInstaller.close();
                         }
+
+                        const wxPlatformInfo& info = wxPlatformInfo::Get();
+                        if ((info.GetOperatingSystemId() & wxOS_WINDOWS) ||
+                            info.GetOperatingSystemId() & wxOS_MAC)
+                        {
+                            if (wxFileName(mAudacityInstallerPath).Exists())
+                                wxTheApp->CallAfter([this] { wxExecute(mAudacityInstallerPath, wxEXEC_ASYNC); });
+                        }
```

Comment:
What if there was an error downloading?  (See other comment about missing check.)

If you know there WAS an error, then you should not make this call.

But if there WAS an error, is your callback called or not?  If you callback is called even in case of error, how do you know if there was one?

It is not clear to me from the class CurlReponse how this is communicated to the caller.  So ask crisp.